experimental/segmenter/src/provider.rs

```diff
@@ -88,3 +88,158 @@ impl Default for LineBreakRuleTable<'static> {
         }
     }
 }
+
+/// Pre-processed Unicode data in the form of tables to be used for rule-based breaking.
+#[icu_provider::data_struct(
+    WordBreakDataV1Marker = "segmenter/word@1",
+    GraphemeClusterBreakDataV1Marker = "segmenter/grapheme@1",
+    SentenceBreakDataV1Marker = "segmenter/sentence@1"
+)]
+#[derive(Debug, PartialEq, Clone)]
+#[cfg_attr(
+    feature = "provider_serde",
+    derive(serde::Serialize, serde::Deserialize)
+)]
+pub struct RuleBreakDataV1<'data> {
+    /// Property table for rule-based breaking.
+    #[cfg_attr(feature = "provider_serde", serde(borrow))]
+    pub property_table: RuleBreakPropertyTable<'data>,
+
+    /// Break state table for rule-based breaking.
+    #[cfg_attr(feature = "provider_serde", serde(borrow))]
+    pub break_state_table: RuleBreakStateTable<'data>,
+
+    /// Number of properties; should be the square root of the length of [`Self::break_state_table`].
+    pub property_count: usize,
+
+    pub last_codepoint_property: i8,
+    pub sot_property: u8,
+    pub eot_property: u8,
+    pub complex_property: u8,
+}
+
+/// Property table for rule-based breaking.
+#[derive(Debug, PartialEq, Clone, yoke::Yokeable)]
+#[cfg_attr(
+    feature = "provider_serde",
+    derive(serde::Serialize, serde::Deserialize)
+)]
+pub enum RuleBreakPropertyTable<'data> {
+    #[cfg_attr(feature = "provider_serde", serde(borrow))]
+    Borrowed(&'data [&'data [u8; 1024]; 897]),
+    Owned(Box<[&'data [u8; 1024]; 897]>),
```

Comment:
Issue: There should be no `'data` in the `Owned` version. See my PR from December where I had to "flatten" the table so that it doesn't contain references.

Reply:
I've made the request change. However, I notice this can increases the size of `RuleBreakDataV1::break_state_table` quite a lot. The table used to contain a lot of duplicate references of `&BREAK_PROPERTIES_FILL_BY_0` , but now it contains duplicate arrays of `BREAK_PROPERTIES_FILL_BY_0`. Line breaker data has the same issue after being fattened. 

I haven't come up with an idea to fixed this yet ...

Reply:
As a follow-up, we may be able to leverage @robertbastian's [deduplicating_array](https://unicode-org.github.io/icu4x-docs/doc/deduplicating_array/index.html) crate if it's useful.